### ISSUE TITLE
[Bug]: xpu_mla_sparse NaN-poisons attention output when a row's leading topk index chunk is fully masked

### ISSUE THREAD

**Issue by nickus** [vllm-project/vllm#48364]
### Your current environment

The defect is in device-agnostic Triton source (`vllm/v1/attention/ops/xpu_mla_sparse.py`), analyzed on vLLM main @ `54503ec`. Reproduced with Triton 3.7.1 — the repro below runs on any GPU, and even under `TRITON_INTERPRET=1` on CPU, since it is a pure numerics bug, not a device one.

### 🐛 Describe the bug

`_bf16_mla_sparse_kernel` NaN-poisons its output whenever the **first `BLOCK_N` (=16) topk index entries of a row are all masked** (`-1` padding or `>= seq_kv`), even though valid keys follow later in the index list.

**Mechanism** (`vllm/v1/attention/ops/xpu_mla_sparse.py`):

- L76: the running max starts at `-inf`: `e_max = tl.zeros([BLOCK_H], dtype=tl.float32) - float("inf")`
- L125: a fully-masked chunk sets every logit to `-inf`: `qk = tl.where(mask, qk, -float("inf"))`
- L139–141:
  ```python
  n_e_max = tl.maximum(tl.max(qk, 1), e_max)   # max(-inf, -inf) = -inf
  re_scale = tl.exp2(e_max - n_e_max)          # exp2(-inf - -inf) = exp2(nan) = nan
  p = tl.exp2(qk - n_e_max[:, None])           # nan
  ```
  `acc *= re_scale` and `e_sum = e_sum * re_scale + tl.sum(p, 1)` are then permanently poisoned — `nan × 0 = nan`, so no later valid chunk can rescue the accumulator. The NaNs flow silently into the attention output.

A row with **no valid key at all** likewise returns NaN (`0/0` at the epilogue), where the in-tree reference (`reference_mla_sparse_prefill` in `tests/kernels/attention/test_xpu_mla_sparse.py`, L67–69) explicitly defines the output of such rows as **zeros**.

**Blast radius** — every caller of `triton_bf16_mla_sparse_interface`:
- `vllm/v1/attention/backends/mla/xpu_mla_sparse.py` L219 (XPU_MLA_SPARSE backend forward)
- `vllm/models/deepseek_v4/xpu/xpu_sparse.py` L342 (DSA prefill)
- `vllm/models/deepseek_v4/xpu/xpu_sparse_decode_fp8.py` L282 (fp8 decode wrapper)

**Why the existing test doesn't catch it:** `test_bf16_triton_sparse_mla` writes valid indices at the *front* of each row and pads with `-1` at the *back* (L98–102), so a leading chunk is never fully masked. Trailing padding is benign (by then `e_max` is finite). The newer sparse-MLA kernel from #47629 already avoids this exact pattern with a finite sentinel; this kernel predates that.

### Minimal repro

```python
# Run on any GPU; runs on CPU too via the Triton interpreter (auto-enabled).
import os
import torch
if not torch.cuda.is_available():
    os.environ["TRITON_INTERPRET"] = "1"
from vllm.v1.attention.ops.xpu_mla_sparse import triton_bf16_mla_sparse_interface

torch.manual_seed(0)
dev = "cuda" if torch.cuda.is_available() else "cpu"
T, HQ, DQK, DV, TOPK = 2, 16, 576, 512, 64        # TOPK % BLOCK_N(16) == 0
q  = torch.randn(T, HQ, DQK, device=dev)
kv = torch.randn(T, 1,  DQK, device=dev)          # seq_kv = 2
idx = torch.full((T, 1, TOPK), -1, dtype=torch.int32, device=dev)
idx[0, 0, 16] = 0   # token 0: its only valid index sits AFTER the first BLOCK_N chunk
idx[1, 0, 0]  = 0   # token 1: valid index inside the first chunk (control)

out, max_logits, lse = triton_bf16_mla_sparse_interface(q, kv, idx, sm_scale=DQK ** -0.5)
print("token0 finite:", bool(out[0].isfinite().all()))   # False  <-- BUG
print("token1 finite:", bool(out[1].isfinite().all()))   # True
```

Output on main @ `54503ec`:

```
token0 finite: False   # all-NaN output despite a valid attention target
token1 finite: True
```

### Fix

Replace `-inf` with a finite sentinel (`-1.0e30`) for the running-max init and the masked logits — the same approach the newer sparse-MLA kernel in #47629 uses. Mixed chunks are unaffected (`exp2(-1e30 - m)` is exactly `0.0` in fp32); leading fully-masked chunks get rescaled away exactly; rows with no valid key produce zeros, matching the reference. Two lines, no perf change. PR with a regression test incoming.
